pxr/usd/usdShade/validators.cpp

```diff
@@ -192,13 +193,45 @@ _ShaderPropertyTypeConformance(const UsdPrim &usdPrim)
     return errors;
 }
 
+
+static
+UsdValidationErrorVector
+_MaterialBindingApiChecker(const UsdPrim &usdPrim)
+{
+    UsdValidationErrorVector errors;
+
+    if (!usdPrim.CanApplyAPI<UsdShadeMaterialBindingAPI>()){
+        return errors;
+    }
+
+    UsdShadeMaterialBindingAPI materialBindingAPI(usdPrim);
+
+    if (materialBindingAPI.GetDirectBinding().IsBound() && !usdPrim.HasAPI<UsdShadeMaterialBindingAPI>()) {
```

Comment:
Hmm.. this is only checking for direct binding, but a prim could also provide a collection based binding rel, which I think it will not cover.

Might be simpler to do the exact same thing which is doing in the python code.. that is, get all relationships in "materialBinding" namespace / startswith, if exists and API not applied report error.